packages/liveblocks-core/src/http-client.ts

```diff
@@ -179,12 +1575,14 @@ export class HttpClient {
    */
   public async post<T extends JsonObject>(
     endpoint: URLSafeString,
+    authCallback: () => Promise<AuthValue>,
```

Comment:
As I shared before, I don't really like this approach of passing an auth callback to _every_ method here. To me, it defeats the purpose of having a (single) auth callback (and in a way a good part of the purpose of this HttpClient abstraction).

At this point, if you need to pass a callback into every call, you might as well just pass the `AuthValue` as a param directly and let the call sites figure it out. The reason I added this HttpClient is to avoid call sites from having to think/know about auth.

So I still think there is some value in trying to assign the auth callback in the constructor. If that's not possible for whatever reason, I'd love to understand that reason and I'd be happy to help brainstorm a potential other solution for this.

Reply:
We previously used to have a `HttpClient` objects per room (actually two, each with a different auth callback on the constructor). I wanted to keep a single http client for the whole client and use that client to make API calls but that might have been a premature thought.

```ts
const httpClient2 = new HttpClient(config.baseUrl, fetchPolyfill, () =>
  // TODO: Use the right scope
  await authManager.getAuthValue({
    requestedScope: "comments:read",
    roomId,
  });
);
```

If we still want to keep the auth callback on the constructor, we could have a map of `HttpClient` by room but not sure if that would make understanding things difficult. I'd love to discuss more on how we can get the best of all worlds.

Reply:
Would it work if we had the following setup?

1. A generic `HttpClient` with an auth callback in the constructor.
2. Have a way of creating a new `HttpClient` from an existing one, using the builder pattern, e.g. `httpClient.withAuth(newCallback)`, returning a new HttpClient instance with a different auth callback.
3. Assign those to each room when it's created.

This way, you end up with an HttpClient instance per room (or one for all rooms), and you still reap the benefits: call sites using the `httpClient` never have to think/know/worry about auth. And which auth method is used by which httpClient instance is also still easy to follow: just look at where the value is created. That's where the auth callback lives.

Let's discuss this idea more in-depth on our call later today 👍